core/commons/che-core-commons-inject/src/main/java/org/eclipse/che/inject/CheBootstrap.java

```diff
@@ -239,11 +243,16 @@ protected void bindProperties(String prefix, Map<String, String> properties) {
                             if (actual == null) {
                                 actual = System.getenv(name);
                             }
-                            if (actual == null) {
+                            if (!skipUnresolved && actual == null) {
```

Comment:
We can rewrite it to avoid double check actual value
```
if (actual != null) {
    buf.append(actual);
} else if (skipUnresolved) {
    buf.append(pValue.substring(i, j));
    LOG.warn("Environment variable " + name + " cannot be resolved, leaving as is.");
} else {
    throw new ConfigurationException(...);
}
```
WDYT?